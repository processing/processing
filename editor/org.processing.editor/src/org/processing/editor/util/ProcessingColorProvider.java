/*******************************************************************************
 * Copyright (c) 2000, 2005 IBM Corporation and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     IBM Corporation - initial API and implementation
 *******************************************************************************/
package org.processing.editor.util;


import java.util.HashMap;
import java.util.Iterator;
import java.util.Map;

import org.eclipse.swt.graphics.Color;
import org.eclipse.swt.graphics.RGB;
import org.eclipse.swt.widgets.Display;

/**
 * Manager for colors used in the Java editor
 */
public class ProcessingColorProvider {

	public static final RGB COMMENT1= new RGB(126, 126, 126); //comment 1
	public static final RGB COMMENT2= new RGB(126, 126, 126); //comment 2
	public static final RGB KEYWORD1= new RGB(204, 102, 0);
	public static final RGB KEYWORD2= new RGB(204, 102, 0);
	public static final RGB KEYWORD3= new RGB(204, 102, 0);
	public static final RGB LITERAL1= new RGB(0, 102, 153); // currently unused [lonnen] june 16, 2010
	public static final RGB LITERAL2= new RGB(0, 102, 153);
	public static final RGB LABEL= new RGB(0, 0, 128); // use listed as '?' in p5 doc
	public static final RGB OPERATOR= new RGB(0, 0, 0);
	public static final RGB INVALID= new RGB(126, 126, 126); // never used [lonnen] june 16, 2010
	public static final RGB STRING= new RGB(0, 102, 153);   
	public static final RGB DEFAULT= new RGB(0, 0, 0);
	
	// used to color JavaDoc by org.processing.editor.javadoc.JavaDocScanner
	// currently ignored, as JavaDoc is treated like normal multiline comments
	// left available to avoid compiler errors from the JavaDocScanner.java class
	// which is never instantiated but wants access to them anyway because the
	// compiler doesn't realize these things. [lonnen] june 16, 2010
	public static final RGB JAVADOC_KEYWORD= new RGB(126, 126, 126);
	public static final RGB JAVADOC_TAG= new RGB(126, 126, 126);
	public static final RGB JAVADOC_LINK= new RGB(126, 126, 126);
	public static final RGB JAVADOC_DEFAULT= new RGB(126, 126, 126);
	
	protected Map fColorTable= new HashMap(17);

	/**
	 * Release all of the color resources held onto by the receiver.
	 */	
	public void dispose() {
		Iterator e= fColorTable.values().iterator();
		while (e.hasNext())
			 ((Color) e.next()).dispose();
	}
	
	/**
	 * Return the color that is stored in the color table under the given RGB
	 * value.
	 * 
	 * @param rgb the RGB value 
	 * @return the color stored in the color table for the given RGB value
	 */
	public Color getColor(RGB rgb) {
		Color color= (Color) fColorTable.get(rgb);
		if (color == null) {
			color= new Color(Display.getCurrent(), rgb);
			fColorTable.put(rgb, color);
		}
		return color;
	}
}
