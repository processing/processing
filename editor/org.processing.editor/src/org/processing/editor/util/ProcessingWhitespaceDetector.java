/*******************************************************************************
 * Copyright (c) 2000, 2005 IBM Corporation and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     IBM Corporation - initial API and implementation
 *******************************************************************************/
package org.processing.editor.util;


import org.eclipse.jface.text.rules.IWhitespaceDetector;

/**
 * A Processing aware white space detector, same as a java aware white space detector.
 */
public class ProcessingWhitespaceDetector implements IWhitespaceDetector {

	/* (non-Javadoc)
	 * Method declared on IWhitespaceDetector
	 */
	public boolean isWhitespace(char character) {
		return Character.isWhitespace(character);
	}
}
