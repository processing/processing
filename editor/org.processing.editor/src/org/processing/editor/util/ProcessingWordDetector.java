/*******************************************************************************
 * Copyright (c) 2000, 2005 IBM Corporation and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     IBM Corporation - initial API and implementation
 *******************************************************************************/
package org.processing.editor.util;


import org.eclipse.jface.text.rules.IWordDetector;

/**
 * A Processing aware word detector. Same as a java aware word detector.
 */
public class ProcessingWordDetector implements IWordDetector {

	/* (non-Javadoc)
	 * Method declared on IWordDetector.
	 */
	public boolean isWordPart(char character) {
		return Character.isJavaIdentifierPart(character);
	}
	
	/* (non-Javadoc)
	 * Method declared on IWordDetector.
	 */
	public boolean isWordStart(char character) {
		return Character.isJavaIdentifierStart(character);
	}
}
