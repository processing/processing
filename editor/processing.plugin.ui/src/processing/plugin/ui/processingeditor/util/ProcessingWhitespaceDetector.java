package processing.plugin.ui.processingeditor.util;

import org.eclipse.jface.text.rules.IWhitespaceDetector;

public class ProcessingWhitespaceDetector implements IWhitespaceDetector {

		public boolean isWhitespace(char character){
			return Character.isWhitespace(character);
		}
	
}
