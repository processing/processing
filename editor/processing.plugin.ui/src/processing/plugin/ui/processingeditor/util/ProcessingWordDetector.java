/**
 * Copyright (c) 2010 Chris Lonnen. All rights reserved.
 *
 * This program and the accompanying materials are made available under the 
 * terms of the Eclipse Public License v1.0 which accompanies this distribution, 
 * and is available at http://www.opensource.org/licenses/eclipse-1.0.php
 * 
 * Contributors:
 *     Chris Lonnen - initial API and implementation
 */
package processing.plugin.ui.processingeditor.util;

import org.eclipse.jface.text.rules.IWordDetector;

/** 
 * Detects Processing words
 * <p>
 * Processing Words are Java words, so this class wraps Character 
 * methods to detect Java words. The interface prevents these
 * from being made static.
 */
public class ProcessingWordDetector implements IWordDetector {

	public boolean isWordPart(char character) {
		return Character.isJavaIdentifierPart(character);
	}
	
	public boolean isWordStart(char character) {
		return Character.isJavaIdentifierStart(character);
	}
}
