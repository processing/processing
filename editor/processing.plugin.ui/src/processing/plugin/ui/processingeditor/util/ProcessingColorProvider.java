/**
 * Copyright (c) 2010 Chris Lonnen. All rights reserved.
 *
 * This program and the accompanying materials are made available under the 
 * terms of the Eclipse Public License v1.0 which accompanies this distribution, 
 * and is available at http://www.opensource.org/licenses/eclipse-1.0.php
 * 
 * Contributors:
 *     Chris Lonnen - initial API and implementation
 */
package processing.plugin.ui.processingeditor.util;

import java.util.HashMap;
import java.util.Iterator;
import java.util.Map;

import org.eclipse.swt.graphics.Color;
import org.eclipse.swt.graphics.RGB;
import org.eclipse.swt.widgets.Display;

/** Manages the colors used in the Processing Editor */
public class ProcessingColorProvider {

	public static final RGB COMMENT1= new RGB(126, 126, 126); //comment 1
	public static final RGB COMMENT2= new RGB(126, 126, 126); //comment 2
	public static final RGB KEYWORD1= new RGB(204, 102, 0);
	public static final RGB KEYWORD2= new RGB(204, 102, 0);
	public static final RGB KEYWORD3= new RGB(204, 102, 0);
	public static final RGB LITERAL1= new RGB(0, 102, 153); // currently unused [lonnen] june 16, 2010
	public static final RGB LITERAL2= new RGB(0, 102, 153);
	public static final RGB LABEL= new RGB(0, 0, 128); // use listed as '?' in p5 doc
	public static final RGB OPERATOR= new RGB(0, 0, 0);
	public static final RGB INVALID= new RGB(126, 126, 126); // never used [lonnen] june 16, 2010
	public static final RGB STRING= new RGB(0, 102, 153);   
	public static final RGB DEFAULT= new RGB(0, 0, 0);
	
	protected Map fColorTable= new HashMap(12); // number of color categories

	/** Release all of the color resources held onto by the receiver. */
	public void dispose() {
		Iterator e= fColorTable.values().iterator();
		while (e.hasNext())
			 ((Color) e.next()).dispose();
	}
	
	/**
	 * Convert an RGB value to a Color using the resource table.
	 * <p>
	 * Please note that this is an SWT color, not a Processing color.
	 */
	public Color getColor(RGB rgb) {
		Color color= (Color) fColorTable.get(rgb);
		if (color == null) {
			color= new Color(Display.getCurrent(), rgb);
			fColorTable.put(rgb, color);
		}
		return color;
	}
	
}
