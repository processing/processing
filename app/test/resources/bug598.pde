// java 5 torture test

import static java.lang.Math.tanh;
import java.util.concurrent.Callable;
import java.util.List;
import java.util.Comparator;
import java.util.Map;
import java.util.Collection;
import java.util.Arrays;
import java.util.HashSet;


private static Comparator<String> rotarapmoc = new Comparator<String>() {
  public int compare(final String o1, final String o2)
  {
    return o1.charAt(o1.length() - 1) - o2.charAt(o2.length() - 1);
  }
};

final <T> void printClass(T t) {
   println(t.getClass()); 
}
final List<String> sortem(final String... strings) {
   Arrays.sort(strings,  rotarapmoc);
   return Arrays.asList(strings);
}

final Map<String, Collection<Integer>>
charlesDeGaulle = new HashMap<String, Collection<Integer>>();

void setup() {
  charlesDeGaulle.put("banana", new HashSet<Integer>());
  charlesDeGaulle.get("banana").add(0);
  System.out.println(sortem("aztec", "maya", "spanish", "portuguese"));
  printClass(12.d);
}
