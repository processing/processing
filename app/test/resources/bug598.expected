import processing.core.*; 

import static java.lang.Math.tanh; 
import java.util.concurrent.Callable; 

import java.applet.*; 
import java.awt.Dimension; 
import java.awt.Frame; 
import java.awt.event.MouseEvent; 
import java.awt.event.KeyEvent; 
import java.awt.event.FocusEvent; 
import java.awt.Image; 
import java.io.*; 
import java.net.*; 
import java.text.*; 
import java.util.*; 
import java.util.zip.*; 
import java.util.regex.*; 

public class bug598 extends PApplet {

// java 5 torture test




private static Comparator<String> rotarapmoc = new Comparator<String>() {
  public int compare(final String o1, final String o2)
  {
    return o1.charAt(o1.length() - 1) - o2.charAt(o2.length() - 1);
  }
};

final <T> void printClass(T t) {
   println(t.getClass()); 
}
public final List<String> sortem(final String... strings) {
   Arrays.sort(strings,  rotarapmoc);
   return Arrays.asList(strings);
}

final Map<String, Collection<Integer>>
charlesDeGaulle = new HashMap<String, Collection<Integer>>();

public void setup() {
  charlesDeGaulle.put("banana", new HashSet<Integer>());
  charlesDeGaulle.get("banana").add(0);
  System.out.println(sortem("aztec", "maya", "spanish", "portuguese"));
  printClass(12.d);
}
    static public void main(String args[]) {
        PApplet.main(new String[] { "--bgcolor=null", "bug598" });
    }
}
